app/api/dao/user.py

```diff
@@ -357,11 +355,11 @@ def get_achievements(user_id):
         """
         user = UserModel.find_by_id(user_id)
         all_relations = user.mentor_relations + user.mentee_relations
-        tasks = []
-        for relation in all_relations:
-            tasks += relation.tasks_list.tasks
-        achievements = [task for task in tasks if task.get("is_done")]
-        return achievements
+        # [relation.task_list...] returns an array with an array of array,
+        # that's why we need to take the first element
+        tasks = [relation.tasks_list.tasks for relation in all_relations][0]
```

Comment:
[0] can crash if list is empty, no?

Reply:
I think no. It returns a list of a list of a list. just have a look at the output:
`[[{'id': 1, 'description': 'Test Task', 'is_done': True, 'created_at': 1578686684.408592, 'completed_at': 1581105884.408592}, {'id': 2, 'description': 'Test Task 2', 'is_done': True, 'created_at': 1578686684.408592, 'completed_at': 1581105884.408592}]]
`
so if this would be empty, it will return `[[]]`, so the frist(0) element would be an empty array, .

Reply:
But as it doesn't really optimize the code, as @sara-02 mentioned, I've removed this change

Reply:
`tasks = sum(relation.tasks_list.tasks for relation in all_relations)` not sure if we can not do this

Reply:
Are we supposed to return the list of achievements or only the number of achievements, we can use your idea in case of the latter?

Reply:
As per the doc string ` Achievements are completed tasks`

Reply:
I think we should not change the functionality in such a code refactor so it should return what it used to return.

Reply:
@annabauza @sara-02 AFAIK, the sum() function just adds all the elements to a list, but it cannot add lists. It returns an error:
`TypeError: unsupported operand type(s) for +: 'int' and 'list'`
I am trying to find a solution right now :)

Reply:
I am hoping the frontend should be doing the task of aggregating the achievement counts, if not then we can propose a minor change in functionality(in a separate PR) to return the achievement count along with the list.

Reply:
> @annabauza @sara-02 AFAIK, the sum() function just adds all the elements to a list, but it cannot add lists. It returns an error:
> `TypeError: unsupported operand type(s) for +: 'int' and 'list'`
> I am trying to find a solution right now :)

Oh my bad, I thought here the sum would return the value of sum of elements of the list :P

Reply:
let's try with something like this `sum_lists = lambda list_a, list_b : list_a + list_a`